enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,206 @@
+
+In this proposal, we will design bundle upgrade for case 1. One of the crucial assumptions we are making in this proposal is that an operator is already running in the cluster which leads to the assumption that an index image already exists that has the from version of the operator bundle. Therefore, we don't create a new index image but use the existing index image (which is a required flag) for the operator and provide the new bundle image for the upgrade. So the previously installed operator will have a subscription (which should be provided by the user) using which we will implement the logic to get the catalog source the existing operator was installed in and dynamically inject the new bundle image into the index image for OLM to detect this change and upgrade to the new version we provide. 
+
+Let's consider a few scenarios: 
+
+  * Verifying operator upgrades
+    * To verify whether an operator upgrade to a desired version has been successful, the ClusterServiceVersion (CSV) CustomResource is a useful resource for providing such context to the end user. A given subscription has its status field and corresponding InstallPlan. If a ClusterServiceVersion (CSV) is created for that InstallPlan, then the CSV Status indicates that the operator has been successfully upgraded. We will ensure that the CSV status is `Successful` and that the previous CSV has been deleted.
+
+  * Do we cleanup resources after failures? 
+    * We keep the registry pod and not clean up any resources as it could be useful for further debugging. Alternative, we could offer flexibility to the operator developers to use a flag --skip-cleanup to indicate whether or not cleanup of resources is desired. This could enable the operator developers to capture all the logs and inspect the pod failures.
+
+  * Behavior of `run bundle-upgrade` in the presence of an OperatorGroup
+    * If an operator is installed, it should have a Subscription and an OperatorGroup already created. This OperatorGroup exists with the correct namespacing and it satifies the subscription. So `run bundle-upgrade` will use the same OperatorGroup that was created during install. However, if the upgrade of the operator doesn't work with the existing OperatorGroup, then error out. 
+    In `run bundle`, there is support for 3 modes: 
+        * AllNamespaces
+        * OwnNamespace
+        * SingleNamespace=ns1
+
+Example UX Scenarios are discussed below. 
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle-upgrade (alias: bu) <bundle-image> --index-image=<> --subscription-name=<> --namespace=<> [--mode=<>]
+```
+
+* Positional arguments:
+    * `bundle-image`
+* Required flags:
+    * `--index-image` 
+    * `--subscription-name`
```

Comment:
Since this is required, we should have `run bundle` print this name after returning @jmrodri.

Reply:
@estroz that makes sense